include/cantera/transport/TransportFactory.h

```diff
@@ -96,6 +96,25 @@ class TransportFactory : public Factory<Transport>
 Transport* newTransportMgr(const std::string& transportModel = "",
                            ThermoPhase* thermo = 0, int loglevel = 0);
 
+//!  Create a new Transport instance.
+/*!
+ *  @param model    name of transport model; if "default", the default
+ *                  transport model for the ThermoPhase object is created
+ *  @param thermo   the ThermoPhase object associated with the phase
+ *  @returns a Transport object for the phase
+ * @ingroup tranprops
+ */
+inline shared_ptr<Transport> newTransport(const std::string& model = "",
+                                          ThermoPhase* thermo = 0) {
```

Comment:
The introduction of this function might be a good time to flip the arguments here around to make more sense - the `thermo` model is required, so it should come before an optional parameter.

Also, wouldn't it make more sense for the default value for `model` to be `"default"`, rather than constructing the "no transport" model?

Reply:
This isn't necessarily how it is currently used, see `_cantera.pxd`:
```
    cdef shared_ptr[CxxTransport] newTransport(string) except +translate_exception
    cdef shared_ptr[CxxTransport] newTransport(string, CxxThermoPhase*) except +translate_exception
```
In essence, the `newTransport("")` is how what was formerly `NoTransport` is instantiated. The class serves as a safe-guard against segfaults (i.e. there will always be a `Transport` model, albeit one that just handles exceptions). This ensures that users see exceptions that point them in the right direction.

Reply:
Yes, being able to create an empty `Transport` object that just raises exceptions makes perfect sense. All I'm suggesting is reorganizing the input arguments, since this is a new function and it's used is still flexible. Currently, it is only ever called with both arguments, from within the Python module.

I think the signature `newTransport(ThermoPhase* thermo, const std::string& model="default")` would provide the simplest interface for the two most common use cases, e.g. creating either the transport model specified by the input file, or a value specified in the call:
```c++
newTransport(phase);
newTransport(phase, "mixture-averaged");
```
This still allows creating an "empty" `Transport` manager with
```c++
newTransport(phase, "None");
```
The rare case (currently unnecessary, as far as I am aware) where you want to create an empty `Transport` object with no attached phase is still possible, and is the only case where the syntax is slightly awkward:
```c++
newTransport(nullptr, "None");
```

Reply:
Np. Was probably too stuck on the old
```C++
Transport* newTransportMgr(const std::string& transportModel = "",
                           ThermoPhase* thermo = 0, int loglevel = 0);
```
and the associated `TransportFactory::newTransport(const std::string& model, ThermoPhase* thermo, int log_level=0)`. 

I'm not sure about the long-term prospects of the factory service functions that return raw pointers (similar for various `ThermoPhase` and `Kinetics` service functions); once this becomes clear, it may make sense to update some of the function signatures. Not sure whether this should be part of 2.6, so those can be gone for 3.0. I certainly don't want to attach this to this PR, as it should be ready to be merged at this point.